client/tests/test_trial.py

```diff
@@ -1,78 +1,89 @@
 """Tests for Trial."""
 import os
+import shutil
+from pathlib import Path
+from typing import Optional
 from unittest import TestCase
-from qiskit.providers import Backend
 
-from purplecaffeine.trial import Trial
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import XGate
+from qiskit.quantum_info import Operator
 
-from .common import populate_trial, test_setup, test_teardown
+from purplecaffeine import Trial, LocalBackend, BaseBackend as TrialBackend
 
-# pylint: disable=no-member
+
+def dummy_trial(backend: Optional[TrialBackend] = None):
+    """Returns dummy trial for tests.
+
+    Returns:
+        dummy trial
+    """
+    trial = Trial("test_trial", backend=backend)
+    trial.add_metric("test_metric", 42)
+    trial.add_parameter("test_parameter", "parameter")
+    trial.add_circuit("test_circuit", QuantumCircuit(2))
+    trial.add_operator("test_operator", Operator(XGate()))
+    trial.add_text("test_text", "text")
+    trial.add_array("test_array", np.array([42]))
+    trial.add_tag("qiskit")
+    trial.add_tag("test")
+    return trial
```

Comment:
Don't you want to test the `add_artifact()` , just to have every function tested ?

Reply:
Let's not add artifact now as I saw you kept files open, which is not a great thing :) 
We need to think how to handle files more carefully.

Reply:
I'm closing it in the tearDown : https://github.com/IceKhan13/purplecaffeine/blob/main/client/tests/test_trial.py#L77

Reply:
but you storing them open